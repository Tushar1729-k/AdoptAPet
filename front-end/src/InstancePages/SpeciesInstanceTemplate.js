import React from 'react'
import { useState, useEffect } from 'react'
import { Container, Row, Col, Card, ListGroup, Button } from 'react-bootstrap'
import { Link, useHistory } from "react-router-dom";
import PropTypes from 'prop-types'
import CustomMap from '../GeneralPages/GoogleMap'
import YoutubePlayer from '../Components/Youtube';
import axios from 'axios';

const SpeciesInstanceTemplate = ({attributes, fetchPage}) => {

    const [lat, setLat] = useState(0)
    const [lng, setLng] = useState(0)
    const fetchCoords = async () => {
        const res = await axios.get(`https://maps.googleapis.com/maps/api/geocode/json?address=${attributes.origin}&key=AIzaSyCD6X1ARbFEh9eXKDW4EbN-kNIHz-_dlaM`)
        setLat(res.data.results[0].geometry.location.lat)
        setLng(res.data.results[0].geometry.location.lng)
    }
    let history = useHistory();

    const redirect = (type, num, path) => {
        fetchPage(type, num)
        history.push(path)
    }

    useEffect(() => {
        window.onbeforeunload = function() {
            return true;
        };
    
        return () => {
            window.onbeforeunload = null;
        };
    }, [])
    return (
        <div>
            <div style={{paddingLeft: "15vw", paddingRight: "15vw", paddingTop: "4vh"}}>
            <Card style={{ width: '70vw' }}>
            <Card.Body>
                <Card.Title style={{fontSize: '6vh'}}>{attributes.breed}</Card.Title>
                <Card.Subtitle style={{fontSize: '4vh'}} className="mb-2 text-muted">{attributes.species}</Card.Subtitle>
                <Card.Text>
                    <Container>
                        <Row>
                            <ListGroup variant="flush">
                                <ListGroup.Item>Height : {attributes.height}</ListGroup.Item>
                                <ListGroup.Item>Weight : {attributes.weight}</ListGroup.Item>
                                <ListGroup.Item>Colors : {attributes.color}</ListGroup.Item>
                                <ListGroup.Item>Energy : {attributes.energy}</ListGroup.Item>
                                <ListGroup.Item>Lifespan : {attributes.lifespan}</ListGroup.Item>
                                <ListGroup.Item>Intelligence : {attributes.intelligence}</ListGroup.Item>
                                <ListGroup.Item>Child Friendly : {attributes.child_friendly}</ListGroup.Item>
                            </ListGroup>
                        </Row>
                        <Row style={{paddingTop: '2vh'}}>
                            <h4>Description</h4>
                            <p>{attributes.wiki}</p>
                        </Row>
                        <Row style={{paddingTop: '2vh'}}>
                            <h4>Temperament</h4>
                            <p>{attributes.temperament}</p>
                        </Row>
                        <Row style={{paddingTop: '2vh'}}>
                            <h4>Shedding</h4>
                            <p>{attributes.shedding}</p>
                        </Row>
                        <Row style={{paddingTop: '2vh'}}>
                            <h4>Potential Health Issues</h4>
                            <p>{attributes.health}</p>
                        </Row>
                    </Container>
                </Card.Text>
                <Row>
                    <h4>Country of Origin: {attributes.origin}</h4>
                    <div style={{paddingLeft: "27.5vw"}}>
                        <Button onClick={() => fetchCoords()}>Click to see Origin Country</Button>
                    </div>
                    <div style={{width: '100vw', height: '100vh'}}>
                    <CustomMap googleMapURL={`https://maps.googleapis.com/maps/api/js?v=3.exp&libraries=geometry,drawing,places&key=
                                              AIzaSyCD6X1ARbFEh9eXKDW4EbN-kNIHz-_dlaM`}
                        lat={lat}
                        lng={lng}
                        zoom={5}
                        loadingElement={<div style={{height: "100%"}}></div>}
                        containerElement={<div style={{height: "100%"}}></div>}
                        mapElement={<div style={{height: "100%"}}></div>}
                    />
                    </div>
                </Row>
                <Row>
                <div style={{paddingTop: "4vh"}}>
                    <div style={{paddingLeft: "17.5vw"}}>
                        <h4>Click to see some videos of the {attributes.breed}!</h4>
                    </div>
                    <YoutubePlayer searchQuery={attributes.breed}/>
                </div>
                </Row>
                <Row style={{paddingTop: '2vh'}}>
                    <Col>
                    {attributes.pets.length === 0 ? <h4>No adoptable pets available</h4> : 
                        <div>
                        <h4>See some adoptable pets</h4>
                        {attributes.pets.map((pet, idx) => (
                            <Button key={idx} variant="link" style={{textDecoration: 'none'}} onClick={() => redirect("ap", pet.api_id, `/apmodel/${pet.api_id}`)}>
                                <h5>Take a look at {pet.name}</h5>
                            </Button>
                        ))}
                        </div>
                    }
                    </Col>
                    <Col>
                    {attributes.adoptCenters.length === 0 ? <h4>No adoption centers cuurrently carry this breed</h4> : 
                        <div>
                            <h4>Adoption Centers that carry this breed</h4>
                                {attributes.adoptCenters.map((center, idx) => (
                                    <Button key={idx} variant="link" style={{textDecoration: 'none'}} onClick={() => redirect("ac", center.api_id, `/acmodel/${center.api_id}`)}>
                                        <h5>{center.name}</h5>
                                    </Button>
                                ))}
                        </div>
                    }
                    </Col>
                </Row>
            </Card.Body>
            </Card>
            </div>
        </div>
    )
}

// Set defaults of props here.
SpeciesInstanceTemplate.defaultProps = {
    attributes: { breed: '', species: '', height: 'NA', weight: 'NA', color: 'NA',
                  energy: 'NA', lifespan: '', temperament: 'NA', shedding: 0,
                  health: 0, description: 'NA', intelligence: 'NA', child_friendly: '',
                  adoptCenters: [], pets: [], wiki: '', origin: ''}
}
// Set type of the prop here.
SpeciesInstanceTemplate.propTypes = {
    attributes: PropTypes.object,
    fetchPage: PropTypes.func
}

export default SpeciesInstanceTemplate
