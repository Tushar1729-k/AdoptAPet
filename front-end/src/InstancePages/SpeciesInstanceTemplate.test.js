import React from 'react';
import SpeciesInstanceTemplate from './SpeciesInstanceTemplate';
import renderer from 'react-test-renderer';

test("Species Instance Page Properly Rendered", () => {
    const component = renderer.create(
        <SpeciesInstanceTemplate
            attributes={{ breed: "German Sheperd", species: "Dog", weight: "10 - 15", height: "", energy: "High",
            color: 'White, black, blue, red, cream and silver, plus various patterns and shadings', lifespan: "10 - 15", 
            temperament: "Angry",
            shedding: 2,
            health: 2,
            origin: "Germany",
            adoptCenters: [{api_id: 12, name: 'Cat Rescue of Maryland, Inc.'}], pets: [{api_id: 1, name: 'Jack'}, {api_id: 2, name: 'John'}], wiki: ""}}
        />
      );
      let tree = component.toJSON();
      expect(tree).toMatchSnapshot();
})