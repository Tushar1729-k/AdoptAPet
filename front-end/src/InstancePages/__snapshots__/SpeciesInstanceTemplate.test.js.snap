// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Species Instance Page Properly Rendered 1`] = `
<div>
  <div
    style={
      Object {
        "paddingLeft": "15vw",
        "paddingRight": "15vw",
        "paddingTop": "4vh",
      }
    }
  >
    <div
      className="card"
      style={
        Object {
          "width": "70vw",
        }
      }
    >
      <div
        className="card-body"
      >
        <div
          className="card-title h5"
          style={
            Object {
              "fontSize": "6vh",
            }
          }
        >
          German Sheperd
        </div>
        <div
          className="mb-2 text-muted card-subtitle h6"
          style={
            Object {
              "fontSize": "4vh",
            }
          }
        >
          Dog
        </div>
        <p
          className="card-text"
        >
          <div
            className="container"
          >
            <div
              className="row"
            >
              <div
                className="list-group list-group-flush"
                onKeyDown={[Function]}
              >
                <div
                  className="list-group-item"
                  data-rr-ui-event-key={null}
                  onClick={[Function]}
                >
                  Height : 
                  
                </div>
                <div
                  className="list-group-item"
                  data-rr-ui-event-key={null}
                  onClick={[Function]}
                >
                  Weight : 
                  10 - 15
                </div>
                <div
                  className="list-group-item"
                  data-rr-ui-event-key={null}
                  onClick={[Function]}
                >
                  Colors : 
                  White, black, blue, red, cream and silver, plus various patterns and shadings
                </div>
                <div
                  className="list-group-item"
                  data-rr-ui-event-key={null}
                  onClick={[Function]}
                >
                  Energy : 
                  High
                </div>
                <div
                  className="list-group-item"
                  data-rr-ui-event-key={null}
                  onClick={[Function]}
                >
                  Lifespan : 
                  10 - 15
                </div>
                <div
                  className="list-group-item"
                  data-rr-ui-event-key={null}
                  onClick={[Function]}
                >
                  Intelligence : 
                </div>
                <div
                  className="list-group-item"
                  data-rr-ui-event-key={null}
                  onClick={[Function]}
                >
                  Child Friendly : 
                </div>
              </div>
            </div>
            <div
              className="row"
              style={
                Object {
                  "paddingTop": "2vh",
                }
              }
            >
              <h4>
                Description
              </h4>
              <p>
                
              </p>
            </div>
            <div
              className="row"
              style={
                Object {
                  "paddingTop": "2vh",
                }
              }
            >
              <h4>
                Temperament
              </h4>
              <p>
                Angry
              </p>
            </div>
            <div
              className="row"
              style={
                Object {
                  "paddingTop": "2vh",
                }
              }
            >
              <h4>
                Shedding
              </h4>
              <p>
                2
              </p>
            </div>
            <div
              className="row"
              style={
                Object {
                  "paddingTop": "2vh",
                }
              }
            >
              <h4>
                Potential Health Issues
              </h4>
              <p>
                2
              </p>
            </div>
          </div>
        </p>
        <div
          className="row"
        >
          <h4>
            Country of Origin: 
            Germany
          </h4>
          <div
            style={
              Object {
                "paddingLeft": "27.5vw",
              }
            }
          >
            <button
              className="btn btn-primary"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              Click to see Origin Country
            </button>
          </div>
          <div
            style={
              Object {
                "height": "100vh",
                "width": "100vw",
              }
            }
          >
            <div
              style={
                Object {
                  "height": "100%",
                }
              }
            />
          </div>
        </div>
        <div
          className="row"
        >
          <div
            style={
              Object {
                "paddingTop": "4vh",
              }
            }
          >
            <div
              style={
                Object {
                  "paddingLeft": "17.5vw",
                }
              }
            >
              <h4>
                Click to see some videos of the 
                German Sheperd
                !
              </h4>
            </div>
            <div>
              <div
                style={
                  Object {
                    "paddingBottom": "2vh",
                    "paddingLeft": "27.5vw",
                  }
                }
              >
                <button
                  className="btn btn-danger"
                  disabled={false}
                  onClick={[Function]}
                  style={
                    Object {
                      "width": "10vw",
                    }
                  }
                  type="button"
                >
                  <svg
                    fill="currentColor"
                    height="1em"
                    viewBox="0 0 16 16"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M0 12V4a2 2 0 0 1 2-2h12a2 2 0 0 1 2 2v8a2 2 0 0 1-2 2H2a2 2 0 0 1-2-2zm6.79-6.907A.5.5 0 0 0 6 5.5v5a.5.5 0 0 0 .79.407l3.5-2.5a.5.5 0 0 0 0-.814l-3.5-2.5z"
                    />
                  </svg>
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          className="row"
          style={
            Object {
              "paddingTop": "2vh",
            }
          }
        >
          <div
            className="col"
          >
            <div>
              <h4>
                See some adoptable pets
              </h4>
              <button
                className="btn btn-link"
                disabled={false}
                onClick={[Function]}
                style={
                  Object {
                    "textDecoration": "none",
                  }
                }
                type="button"
              >
                <h5>
                  Take a look at 
                  Jack
                </h5>
              </button>
              <button
                className="btn btn-link"
                disabled={false}
                onClick={[Function]}
                style={
                  Object {
                    "textDecoration": "none",
                  }
                }
                type="button"
              >
                <h5>
                  Take a look at 
                  John
                </h5>
              </button>
            </div>
          </div>
          <div
            className="col"
          >
            <div>
              <h4>
                Adoption Centers that carry this breed
              </h4>
              <button
                className="btn btn-link"
                disabled={false}
                onClick={[Function]}
                style={
                  Object {
                    "textDecoration": "none",
                  }
                }
                type="button"
              >
                <h5>
                  Cat Rescue of Maryland, Inc.
                </h5>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
